finance-service/src/main/kotlin/com/ajaxproject/financeservice/service/FinanceService.kt

```diff
@@ -1,36 +1,42 @@
 package com.ajaxproject.financeservice.service
 
 import com.ajaxproject.financeservice.repository.FinanceRepository
-import com.ajaxproject.financemodelsapi.enums.Finance
-import com.ajaxproject.financemodelsapi.enums.Finance.EXPENSE
-import com.ajaxproject.financemodelsapi.enums.Finance.INCOME
-import com.ajaxproject.financemodelsapi.models.MongoFinance
+import com.ajaxproject.financemodels.enums.Finance
+import com.ajaxproject.financemodels.enums.Finance.EXPENSE
+import com.ajaxproject.financemodels.enums.Finance.INCOME
+import com.ajaxproject.financemodels.models.MongoFinance
 import com.ajaxproject.internalapi.finance.commonmodels.FinanceMessage
 import com.ajaxproject.internalapi.finance.commonmodels.FinanceType
-import org.bson.types.ObjectId
 import org.springframework.stereotype.Service
+import reactor.core.publisher.Mono
+import reactor.kotlin.core.publisher.toMono
 import java.util.*
 
 @Service
 class FinanceService(
     private val financeRepositoryImpl: FinanceRepository,
 ) {
 
-    fun getAllFinancesByUserId(userId: Long, financeType: Finance): List<MongoFinance> {
+    fun getAllFinancesByUserId(userId: Long, financeType: Finance): Mono<List<MongoFinance>> {
         return financeRepositoryImpl.findByUserIdAndFinanceType(userId, financeType)
+            .collectList()
     }
 
-    fun addFinance(finance: MongoFinance): MongoFinance {
+    fun addFinance(finance: MongoFinance): Mono<MongoFinance> {
         return financeRepositoryImpl.save(finance)
     }
 
-    fun deleteFinanceByUserId(id: ObjectId) {
-        financeRepositoryImpl.deleteById(id)
+    fun removeAllFinancesByUserId(userId: Long): Mono<Unit> {
+        return Mono.just(financeRepositoryImpl.removeAllById(userId))
```

Comment:
```suggestion
        return financeRepositoryImpl.removeAllById(userId)
```
your repository should be reactive